### ISSUE TITLE
Update `@yamada-ui/breadcrumb` accessibility

### ISSUE THREAD

**Issue by setodeve** [yamada-ui/yamada-ui#1726]
### Description
This issue aims to address accessibility deficiencies in our `breadcrumb component` as per W3C guidelines. 
The component currently does not meet the accessibility standards outlined in the W3C documentation. 

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

#### General Requirements
- [x] **Navigation Landmark:**
  - [x] Breadcrumb trail is contained within a `navigation` landmark region.

- [ ] **Landmark Region Label:**
  - [ ] The navigation landmark region is labeled using `aria-label` or `aria-labelledby`.

- [x] **Current Page Indicator:**
  - [x] The link to the current page has `aria-current="page"`.

### Problem Statement/Justification

none

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### References

https://www.w3.org/WAI/ARIA/apg/patterns/breadcrumb/

**Comment by cidkumagai** [comment]
I'll try it.
